File: Venkatesh/Linked-List/LinkedList.java
public class LinkedList {

    Node head;

    class Node {
        int data;
        Node next;

        Node (int data) {
            this.data = data;
            this.next = null;
        }
    }

    public void insertion(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
        } else {
            Node last = head;
            while (last.next != null) {
                last = last.next;
            }
            last.next = newNode;
        }
    }

    public void removeDuplicates() {
        Node temp = head;
        if (temp == null) {
            return;
        }
        while (temp.next != null) {
            if (temp.data == temp.next.data) {
                temp.next = temp.next.next;
            } else {
                temp = temp.next;
            }
        }
    }

    public void printList() {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + (temp.next != null ? " -> " : ""));
            temp = temp.next;
        }
    }

    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        list.insertion(1);
        list.insertion(2);
        list.insertion(2);
        list.insertion(4);
        list.insertion(4);
        list.insertion(4);
        list.insertion(5);

        list.removeDuplicates();
        list.printList();
    }
}

File: Venkatesh/Linked-List/LinkedList.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->next = NULL;
    return node;
}

void insertion(struct Node** head, int data) {
    struct Node* node = newNode(data);
    if (*head == NULL) {
        *head = node;
    } else {
        struct Node* temp = *head;
        while (temp->next != NULL) {
            temp = temp->next;
        }
        temp->next = node;
    }
}

void removeDuplicates(struct Node** head) {
    struct Node* temp = *head;
    while (temp != NULL && temp->next != NULL) {
        if (temp->data == temp->next->data) {
            struct Node* toDelete = temp->next;
            temp->next = temp->next->next;
        } else {
            temp = temp->next;
        }
    }
}

void printList(struct Node* head) {
    struct Node* temp = head;
    while (temp != NULL) {
        printf("%d", temp->data);
        if (temp->next != NULL) printf(" -> ");
        temp = temp->next;
    }
    printf("\n");
}

int main() {
    struct Node* list = NULL;

    insertion(&list, 1);
    insertion(&list, 2);
    insertion(&list, 2);
    insertion(&list, 4);
    insertion(&list, 4);
    insertion(&list, 4);
    insertion(&list, 5);

    removeDuplicates(&list);
    printList(list);
}